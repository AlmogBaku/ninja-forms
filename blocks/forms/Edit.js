import React from 'react';

export default  function Edit() {
    return <React.Fragment/>
}