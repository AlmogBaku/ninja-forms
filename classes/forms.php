<?php
/**
 * Handles adding and removing forms.
 *
 * @package     Ninja Forms
 * @subpackage  Classes/Form
 * @copyright   Copyright (c) 2014, WPNINJAS
 * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
 * @since       2.9
*/

class NF_Forms {

	/**
	 * Get things started
	 * 
	 * @access public
	 * @since 2.9
	 * @return void
	 */
	public function __construct() {
		
	}

	/**
	 * Get all forms
	 * 
	 * @access public
	 * @since 2.9
	 * @return array $forms
	 */
	public function get_all( $debug = false, $show_new = false ) {
		$forms = nf_get_objects_by_type( 'form' );

		$tmp_array = array();
		foreach ( $forms as $form ) {
			$form_id = $form['id'];

			$status = Ninja_Forms()->form( $form_id )->get_setting( 'status' );
			if ( ( $status == 'new' && $show_new ) || $status != 'new' ) {
				$title = Ninja_Forms()->form( $form_id )->get_setting( 'form_title' );
				if ( strpos( $title, '_' ) === 0 ) {
					if ( $debug )
						$tmp_array[] = $form_id;
				} else {
					$tmp_array[] = $form_id;
				}
			}
		}
		return $tmp_array;
	}
}