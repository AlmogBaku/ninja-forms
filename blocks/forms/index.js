import Edit from './Edit';
import  block from './block';
import { __ } from '@wordpress/i18n';
import { registerBlockType } from '@wordpress/blocks';

registerBlockType('ninja-forms/form',{
    ...block,
    title: __('Ninja Form', 'ninja-forms'),
    edit: ({attributes,setAttributes}) => {

        return <Edit/>
    },
    save: () => null
});