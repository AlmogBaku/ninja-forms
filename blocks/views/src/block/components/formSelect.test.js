import forms from '../../data/forms'
import { FormSelect } from './formSelect';
import {
render, //test renderer
cleanup, //resets the JSDOM
fireEvent, //fires events on nodes,
} from "@testing-library/react";

describe("Editor component", () => {
    afterEach(cleanup);

    it("matches snapshot", () => {
        const props = { forms: forms, formId: 1, onChange: jest.fn() }

        const {container} = render(
            <FormSelect {...props} />
        );

        expect(container).toMatchSnapshot();
    });

    it("changes value", () => {
        const onFormChange = jest.fn();
        const props = { forms: forms, formId: 1, onChange: onFormChange }
        const {getByLabelText} = render(
            <FormSelect {...props} />
        )
        
        fireEvent.change(
            getByLabelText('Select a form'),
            {target: { value: 1 }}
        );
        
        expect(onFormChange).toHaveBeenCalledTimes(1);
        expect(onFormChange).toHaveBeenCalledWith("1");
    })

});