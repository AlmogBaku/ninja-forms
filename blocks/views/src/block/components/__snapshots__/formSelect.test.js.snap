// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor component matches snapshot 1`] = `
<div>
  <div
    class="components-base-control"
  >
    <div
      class="components-base-control__field"
    >
      <label
        class="components-base-control__label"
        for="inspector-select-control-0"
      >
        Select Form
      </label>
      <select
        class="components-select-control__input"
        id="inspector-select-control-0"
      >
        <option
          value="0"
        >
          -
        </option>
        <option
          value="1"
        >
          Test Form
        </option>
      </select>
    </div>
  </div>
</div>
`;
